### PR TITLE
More Analyzers and Some Refactoring

### PR DESCRIPTION
**Pull request by matan129**
Essentially:
 * I've added a new analyzer (`strict_partial_network_address`) - accepts MAC/IP-like strings if they have more than one "part" - for example, `AA` will not be analyzed but `AA:BB` will.
 * Made the `IncrementalCaptureGroupTokenFilter` set position attribute and token offsets - useful for highlighting.
 * Fixed some edge cases 
 * Added dockerfile for testing
 * Adapt for new ES versions (although you now cover 6.0 - it wasn't the case when I started the fork)
 * Travis CI support
 * Streamline tests and general refactoring.

Please review carefully 🔍

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+install: true
+sudo: false
+
+before_deploy:
+  - mvn package
+
+deploy:
+  provider: releases
+  api_key:
+    secure: rqnGf6B8JfbocFdV1qfRBVMLjAEbTXz/tdpQ91vl2jIBdcxdSgCarw2qvE85PAsO43P5aNicF6SqGtPzGotcOXkCQZGpIR9en+a6hi7NBClnTsy5jwdxtYL3se8MeA81NtlMOVSi5LgxapNRxCZQa7tHTheKV5MQF3JI/TLW9z0am64xKcdw23/fZKByGGQwsar6ywmtMv3QNKEwud/IlpCgWUIys99Zb/m1nUaiuWBogslt4mxda9reeJAydYYd0f2lQUwTEQfs8w7aAOa7mDAE60r2g2dLEwaHLtrJg17xcqB74Gg39/8kOuEGSIEG8INrPd602XNsHcFRp4cV3+tB6oZXC7dBt/RnllaqujLsu6/z5Zrl8++AtONEXqHxijvxD057XkN/1yQ2YxmI69+QTqDkKspWTX0zqK0Ywxe0TpAjNzxuywV0xjSB9/UMWqg+Sh5ItVmL6Sm1FAO3oC0DmBjjrr5fbZkSRDPPwoi1B8KnJQBmAJkCq70azZL6P3INjJeD0SF1/d9Ww/VWV7uZhUvbA4Ul8QdypvQrk9DYDTxsyFoMf5rbn4eDO/blHiirtv8Fs2vJAMBecmjZZNcITKKxlW+EWTWvgv2RcY/8ZFrSpj9uHQI8JzYsGrZv8GU6bQx4mJPgV2H+Y47dkwK1FWlu5kmkyc6vHjhI6mw=
```
Comment (matan129) [inline .travis.yml]: This will break in your repo for obvious reasons

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+END OF TERMS AND CONDITIONS
+
+APPENDIX: How to apply the Apache License to your work.
+
+To apply the Apache License to your work, attach the following boilerplate notice, with the fields enclosed by brackets "{}" replaced with your own identifying information. (Don't include the brackets!) The text should be enclosed in the appropriate comment syntax for the file format. We also recommend that a file or class name and description of purpose be included on the same "printed page" as the copyright notice for easier identification within third-party archives.
+
+Copyright 2017 https://github.com/matan129/elasticsearch-network-path-analysis/graphs/contributors
```
Comment (matan129) [inline LICENSE]: Will change this in your repo

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+FROM docker.elastic.co/elasticsearch/elasticsearch:5.5.2
```
Comment (matan129) [inline Dockerfile]: I think we have to decide how we backport the repo for different ES versions,
For example, I think that keeping the `master` in the most recent ES version and having different branches for other versions is a good idea. 

When it's decided the dockerfile version should change 👍

File: pom.xml [hunk 4]
```diff
@@ -2,94 +2,81 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-	<description>An Elasticsearch plugin for network address analysis</description>
+    <description>Elasticsearch plugin for network address and path analysis</description>
     <modelVersion>4.0.0</modelVersion>
-    <groupId>com.ofir123</groupId>
-    <artifactId>elasticsearch-network-analysis</artifactId>
-    <version>5.3</version>
+    <groupId>com.matan129</groupId>
```
Comment (matan129) [inline pom.xml]: Maybe a new groupId?
(or I can leave it on your name, it's fine by me)

**Review by matan129 — COMMENTED** [review]
I've written some general remarks I want to resolve before merging.

**Comment by ofir123** [comment]
Thank you very very much @matan129!
I'll go over all of your changes in the next couple of weeks, and will update once the fixes are ready.

File: .travis.yml (hunk 1 above)
Comment (ofir123) [reply .travis.yml]: I've created my own `travis.yml`, so overwrite this one.

File: LICENSE (hunk 2 above)
Comment (ofir123) [reply LICENSE]: Again, overwritten by my change.

File: .travis.yml (hunk 1 above)
Comment (ofir123) [reply .travis.yml]: Overwritten by my change.

File: README.md [hunk 5]
```diff
@@ -1,122 +1,49 @@
-Network Address Analysis for ElasticSearch
+Network Address and Path Analysis for Elasticsearch
```
Comment (ofir123) [inline README.md]: Overwritten by new commit.

File: pom.xml (hunk 4 above)
Comment (ofir123) [reply pom.xml]: `com.elasticsearch.networkanalysis`?

File: src/main/java/org/elasticsearch/index/analysis/BasePatternAnalyzer.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.analysis.Tokenizer;
+import org.apache.lucene.analysis.core.LowerCaseFilter;
+import org.apache.lucene.analysis.pattern.PatternTokenizer;
+import org.elasticsearch.common.regex.Regex;
+import org.elasticsearch.common.settings.Settings;
+
+import java.util.regex.Pattern;
+
+public abstract class BasePatternAnalyzer extends Analyzer {
```
Comment (ofir123) [inline src/main/java/org/elasticsearch/index/analysis/BasePatternAnalyzer.java:13]: Nice!

File: src/main/resources/plugin-metadata/plugin-descriptor.properties [hunk 7]
```diff
@@ -4,6 +4,6 @@ name=${elasticsearch.plugin.name}
 site=${elasticsearch.plugin.site}
 jvm=${elasticsearch.plugin.jvm}
 classname=${elasticsearch.plugin.classname}
-java.version=${java.specification.version}
+java.version=8
```
Comment (ofir123) [inline src/main/resources/plugin-metadata/plugin-descriptor.properties]: Why not use the specs version?

File: src/main/resources/plugin-metadata/plugin-assembly.xml [hunk 8]
```diff
@@ -9,25 +9,25 @@
         <fileSet>
             <directory>${project.basedir}/src/main/plugin-metadata</directory>
             <includes>
-              <include>plugin-security.policy</include>
+                <include>plugin-security.policy</include>
             </includes>
-            <outputDirectory></outputDirectory>
+            <outputDirectory>/elasticsearch</outputDirectory>
```
Comment (ofir123) [inline src/main/resources/plugin-metadata/plugin-assembly.xml:14]: Why the change?

**Review by ofir123 — CHANGES_REQUESTED** [review]
Your changes are all more than welcome!
Thank you very much :)
I've overwritten some of the files you changed, so you'll have some new conflicts.
Can you leave only the Java files in this PR? (we can discuss other changes separately)

**Comment by ofir123** [comment]
If you want @matan129, you can give me edit permissions and I'll merge everything.

File: pom.xml (hunk 4 above)
Comment (matan129) [reply pom.xml]: No, this is too confusing with official ES repos (the ES package are `org.elasticsearch` something).
nvm, let's leave it this way

File: src/main/resources/plugin-metadata/plugin-assembly.xml (hunk 8 above)
Comment (matan129) [reply src/main/resources/plugin-metadata/plugin-assembly.xml:14]: New ES versions require the plugin zip to have it's content in a dir called `elasticsearch`.
Proof that it won't work without it:
https://asciinema.org/a/LtxZyF5gAuoxX9cwFQvHsNoTQ

**Comment by matan129** [comment]
I've pulled your (upstream) changes, now the PR should only have code + the plugin assembly dir change.

Btw, I'd love write permissions on this repo, since I don't think there is a reason for my fork to exist after this merge.

**Comment by ofir123** [comment]
Sure! Sent you an invite..

**Comment by matan129** [comment]
Cool, thanks.
Any reason this PR is not merged yet (obviously I could merge myself, but..)?

**Comment by ofir123** [comment]
Haven't had a chance to look at everything yet.. sorry!
This weekend I'll merge everything..

**Review by ofir123 — APPROVED** [review]
